tests/execution_engine/test_pandas_execution_engine.py

```diff
@@ -355,6 +355,28 @@ def test_s3_files(s3, s3_bucket, test_df_small_csv):
     return s3_bucket, keys
 
 
+@pytest.fixture
+def test_s3_files_parquet(tmpdir, s3, s3_bucket, test_df_small, test_df_small_csv):
+    keys: List[str] = [
+        "path/A-100.csv",
+        "path/A-101.csv",
+        "directory/B-1.parquet",
+        "directory/B-2.parquet",
+        "alpha-1.csv",
+        "alpha-2.csv",
+    ]
+    path = Path(tmpdir) / "file.parquet"
+    test_df_small.to_parquet(path=path)
+    with open(path, "rb") as infile:
+        parquet = infile.read()
+    for key in keys:
+        if key.endswith(".parquet"):
+            s3.put_object(Bucket=s3_bucket, Body=parquet, Key=key)
```

Comment:
Nit: can just pass the path instead of reading the file
```suggestion
    test_df_small.to_parquet(path=path)
    for key in keys:
        if key.endswith(".parquet"):
            s3.put_object(Bucket=s3_bucket, Body=path, Key=key)
```